### PR TITLE
Issue 5192 - Add Kotlin-DSL example to readme.md

### PR DESCRIPTION
**Pull request by mattdkerr**
To resolve https://github.com/grpc/grpc-java/issues/5192 this adds a "community supported" Kotlin-DSL example. This example is not as nearly as terse as the Groovy example due to current limitations of the protobuf-gradle-plugin and not because of the approach in Kotlin-DSL itself. In the near future, this should be resolved when issue https://github.com/google/protobuf-gradle-plugin/issues/219 is addressed. The general approach here is to use the samples given for ```<details>``` blocks from the Gradle 5.0 documentation.

### CODE REVIEW

**Comment by thelinuxfoundation** [comment]
Thank you for your pull request. Before we can look at your contribution, we need to ensure all contributors are covered by a Contributor License Agreement.

After the following items are addressed, please respond with a new comment here, and the automated system will re-verify.

* User @mattdkerr isn't covered by a CLA. They will need to complete the form at https://identity.linuxfoundation.org/projects/cncf

_Regards,
CLA GitHub bot_

**Comment by mattdkerr** [comment]
I believe @srini100 is assigned to this issue

**Comment by zhangkun83** [comment]
Now that google/protobuf-gradle-plugin#219 is resolved in the [0.8.7 release](https://github.com/google/protobuf-gradle-plugin/releases/tag/v0.8.7), I wonder if we should just upgrade the protobuf-gradle-plugin version in the example and use the [cleaner Kotlin DSL example](https://github.com/google/protobuf-gradle-plugin/blob/v0.8.7/examples/exampleKotlinDslProject/build.gradle.kts) here.

**Comment by ejona86** [comment]
I'll let @zhangkun83 do the initial review, but I do think the snippet should be optimized at least partially for size. If it remains the same approximate size as it is now it shouldn't exist in the README and should become its own file or some such.

**Comment by dapengzhang0** [comment]
Ping @zhangkun83 any update?
>Now that [google/protobuf-gradle-plugin#219](https://github.com/google/protobuf-gradle-plugin/issues/219) is resolved in the [0.8.7 release](https://github.com/google/protobuf-gradle-plugin/releases/tag/v0.8.7), I wonder if we should just upgrade the protobuf-gradle-plugin version in the example and use the [cleaner Kotlin DSL example](https://github.com/google/protobuf-gradle-plugin/blob/v0.8.7/examples/exampleKotlinDslProject/build.gradle.kts) here

**Comment by zhangkun83** [comment]
@mattdkerr, would mind updating your PR according to my earlier comment? Different opinions are welcome.

**Comment by mattdkerr** [comment]
Updated- thanks for the reminder.

File: README.md [hunk 1]
```diff
@@ -123,34 +123,149 @@ For protobuf-based codegen integrated with the Maven build system, you can use
...
+
+buildscript {
+    extra.set("grpcVersion", "1.18.0")
+    extra.set("protocVersion", "3.6.1")
+    extra.set("kotlinVersion", "1.3.20")
+    extra.set("protoGradleVersion", "0.8.8")
+}
+
+plugins {
+    kotlin("jvm") version "1.3.20"
```
Comment (zhangkun83) [inline README.md]: It doesn't seem to be necessary for using Kotlin DSL, does it?

File: README.md [hunk 2]
```diff
@@ -123,34 +123,149 @@ For protobuf-based codegen integrated with the Maven build system, you can use
...
+            srcDir 'src/main/protobuf'
+            srcDir 'src/main/protocolbuffers'
+        }
+        // Optional: compile the generated Java code
+        java {
+            srcDirs(
+                "build/generated/source/proto/main/java",
+                "build/generated/source/proto/main/grpc"
+            )
+        }
```
Comment (zhangkun83) [inline README.md]: Why is this necessary? Generated files are automatically added as the input for Java compile tasks. Is it because it's not the case for Kotlin compile tasks?

File: README.md [hunk 3]
```diff
@@ -123,34 +123,149 @@ For protobuf-based codegen integrated with the Maven build system, you can use
...
+    }
+}
+
+dependencies {
+    // ********************************* Kotlin ******************************************
+    // Use the Kotlin JDK 8 standard library
+    val kotlinVersion : String by rootProject.extra
+    implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlinVersion")
+    implementation("org.jetbrains.kotlin:kotlin-reflect:$kotlinVersion")
```
Comment (zhangkun83) [inline README.md]: These two lines seem for compiling Kotlin sources.

File: README.md [hunk 4]
```diff
@@ -123,34 +123,149 @@ For protobuf-based codegen integrated with the Maven build system, you can use
...
+</details>
+
+A community-supported Kotlin-DSL example is also available:
+<details>
+<summary>Kotlin DSL</summary>
+
+```kotlin
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+import com.google.protobuf.gradle.*
+import org.gradle.kotlin.dsl.provider.gradleKotlinDslOf
```
Comment (TimelyToga) [inline README.md]: What is this import for?

File: README.md [hunk 5]
```diff
@@ -123,34 +123,149 @@ For protobuf-based codegen integrated with the Maven build system, you can use
...
+    mavenCentral()    
+    maven("https://plugins.gradle.org/m2/")
+}
+
+sourceSets {
+    main {
+        // Optional: customize where .proto are found for src/main/
+        proto {
+            // In addition to the default 'src/main/proto'
+            srcDir 'src/main/protobuf'
```
Comment (TimelyToga) [inline README.md]: I think this should be: 

```
srcDir("src/main/protobuf")
```

**Comment by ejona86** [comment]
Closing due to lack of activity by the original author. We'd be happy to reopen if there is something further to review.
